spec/compiler/crystal/tools/init_spec.cr

```diff
@@ -1,149 +1,175 @@
-require "spec"
-require "yaml"
-require "ini"
 require "compiler/crystal/config"
 require "compiler/crystal/tools/init"
+require "file_utils"
+require "ini"
+require "spec"
+require "yaml"
+
+BIN_CRYSTAL = File.expand_path("bin/crystal")
+
+private def exec_init(project_name, project_dir = nil, type = "lib")
+  args = ["init", type, project_name]
+  args << project_dir if project_dir
+
+  process = Process.new(BIN_CRYSTAL, args, shell: true, error: Process::Redirect::Pipe)
+  stderr = process.error.gets_to_end
+  status = process.wait
+  $? = status
+  stderr
+end
+
+# Creates a temporary directory, cd to it and run the block inside it.
+# The directory and its content is deleted when the block return.
+private def within_temporary_directory
+  tmp_path = "/tmp/crystal_init_spec_tmp_dir-#{rand(10_000)}"
```

Comment:
Could `#{__DIR__}/tmp...` be used instead of `/tmp...`? The local tmp is already in the .gitignore and I would rather not write something outside the working copy.

Other that that, please rebase and I would 👍

Reply:
Aah ok, I didn't thought about that, I do that soon!